src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/BaseRunner.cs

```diff
@@ -118,47 +177,56 @@ private async Task<(TestResult, string)> RunAsync2(TestCase testCase)
             {
                 try
                 {
-                    result = await RunAsyncInternal(testCase.Input, testCase.SetupHandlerName);
-                }
-                catch (SetupHandlerNotFoundException)
-                {
-                    return (TestResult.Skip, $"was skipped due to missing setup handler {testCase.SetupHandlerName}");
-                }
-
-                actualStr = TestRunner.TestToString(result);
-            }
-            catch (Exception e)
```

Comment:
>catch (Exception e) [](http://example.com/codeflow?start=12&length=19)

Key difference here is that before we would expect an exception for compile errors (thrown via CheckResult.ThrowOnErrors). 

Now we expect an explicit return result.   And exceptions are truly exceptional.